packages/design-picker/src/components/generated-design-picker.tsx

```diff
@@ -115,14 +118,51 @@ const GeneratedDesignPicker: React.FC< GeneratedDesignPickerProps > = ( {
 	verticalId,
 	locale,
 	heading,
+	footer,
+	isStickyFooter,
 	onPreview,
 	onViewMore,
 } ) => {
 	const { __ } = useI18n();
+	const headingRef = useRef< HTMLDivElement >( null );
+	const observerRef = useRef< IntersectionObserver >();
+	const [ isHeadingInViewport, setIsHeadingInViewport ] = useState( true );
+
+	useEffect( () => {
+		if ( ! headingRef.current || observerRef.current || ! isStickyFooter ) {
+			return;
+		}
+
+		const handler = ( entries: IntersectionObserverEntry[] ) => {
+			if ( ! entries.length ) {
+				return;
+			}
+
+			const [ entry ] = entries;
+			setIsHeadingInViewport( entry.isIntersecting );
+		};
+
+		let buttonEl;
+		for ( const childEl of headingRef.current.children ) {
```

Comment:
I think we can just use `getElementsByTagName` to find that button. What do you think?

Reply:
Changing implementation 🧑🏻‍💻